### ISSUE TITLE
Slack engine error using aliases: TypeError unhashable type

### ISSUE THREAD

**Issue by Giandom** [saltstack/salt#46951]
### Description of Issue/Question
I get the following error trying to run from slack aliases:
```
2018-04-09 09:43:44,803 [salt.engines     :132 ][CRITICAL][7450] Engine 'slack' could not be started!
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/engines/__init__.py", line 128, in run
    self.engine[self.fun](**kwargs)
  File "/usr/lib/python2.7/site-packages/salt/engines/slack.py", line 769, in start
    raise Exception('{}'.format(traceback.format_exc()))
Exception: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/engines/slack.py", line 767, in start
    client.run_commands_from_slack_async(message_generator, fire_all, tag, control)
  File "/usr/lib/python2.7/site-packages/salt/engines/slack.py", line 644, in run_commands_from_slack_async
    for msg in message_generator:
  File "/usr/lib/python2.7/site-packages/salt/engines/slack.py", line 476, in generate_triggered_messages
    data['user_name'], msg_text, loaded_groups, trigger_string)
  File "/usr/lib/python2.7/site-packages/salt/engines/slack.py", line 354, in control_message_target
    use_cmdline = self.commandline_to_list(permitted_group[1]['aliases'][cmdline[0]], '')
  File "/usr/lib/python2.7/site-packages/salt/engines/slack.py", line 309, in commandline_to_list
    cmdline = salt.utils.args.shlex_split(cmdline_str[len(trigger_string):])
TypeError: unhashable type
```

Any advices?

### Setup
```
engines:
  - slack:
      token: 'xoxb-xxxxxxxxxx-xxxxxxxxxxxxxxxxxxx'
      control: True
      fire_all: False
      groups_pillar_name: 'slack_engine:groups_pillar'
      groups:
        my_group:
          users:
            - my_user
          commands:
            - test.ping
            - cmd.run
            - list_jobs
            - list_commands
            - state.apply
          aliases:
            list_jobs:
              cmd: jobs.list_jobs
            list_commands:
              cmd: pillar.get salt:engines:slack:valid_commands target=*master*
```

### Steps to Reproduce Issue
From slack:
```
!list_jobs
```

### Versions Report
```
Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.1
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.17.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

**Comment by Ch3LL** [comment]
was this working on a previous salt version or is this your first setup of slack engine?

**Comment by garethgreenaway** [comment]
@Ch3LL Looking into it

**Comment by rallytime** [comment]
Closed via #47008
